src/azul/azulclient.py

```diff
@@ -376,6 +379,41 @@ def delete_bundle(self, catalog: CatalogName, bundle_uuid, bundle_version):
         ]
         self.index(catalog, notifications, delete=True)
 
+    def deindex(self, catalog: CatalogName, sources: Iterable[str]):
+        plugin = self.repository_plugin(catalog)
+        sources = [plugin.resolve_source(s).id for s in sources]
+        es_client = ESClientFactory.get()
+        indices = ','.join(self.index_service.index_names(catalog))
+        query = {
+            "query": {
+                "bool": {
+                    "should": [
+                        {
+                            "terms": {
+                                "sources.id.keyword": sources
+                            }
+                        },
+                        {
+                            "terms": {
+                                "source.id.keyword": sources
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+        logger.info('Deindexing sources: %s from catalog: %s', sources, catalog)
+        logger.debug('Using query: %s', json.dumps(query, indent=4))
+        response = es_client.delete_by_query(index=indices, body=query, slices='auto')
+        logger.debug(json.dumps(response, indent=4))
+        if response['version_conflicts'] > 0:
+            logger.error('Version conflicts encountered. Do not deindex while '
+                         'indexing is occurring. The index may now be in an '
+                         'inconsistent state.')
```

Comment:
I feel there should be a warning about this in the `reindex.py` script so the user can be made aware of this before starting the operation. Consider copying this warning to the help string for `--deindex`.

Reply:
Also, while it would introduce some significant startup lag, you might also want to consider reusing the `ReindexDetector` class from `scripts/sell_unused_slots.py` (you would need to move the class out of `scripts/` and into `src/azul/` in order to import it). That would let you check whether there was an ongoing reindex operation and prevent the deindex operation from causing conflicts in the first place. Up to you whether to pursue this.

Reply:
Warning added, new ticket created to address the latter: #3284